src/GSvar/RepeatExpansionWidget.cpp

```diff
@@ -471,101 +483,173 @@ void RepeatExpansionWidget::loadMetaDataFromNGSD()
-		setCell(row, "min. pathogenic", min_pathogenic);
-
-		//inheritance
-		QString inheritance = db.getValue("SELECT inheritance FROM repeat_expansion WHERE id=" + id).toString().trimmed();
-		setCell(row, "inheritance", inheritance);
-
-		//location
-		QString location = db.getValue("SELECT location FROM repeat_expansion WHERE id=" + id).toString().trimmed();
-		setCell(row, "location", location);
-
-		//diseases
-		QString disease_names = db.getValue("SELECT disease_names FROM repeat_expansion WHERE id=" + id).toString().trimmed();
-		setCell(row, "diseases", disease_names);
-
-		//OMIM IDs
-		QStringList disease_ids_omim = db.getValue("SELECT disease_ids_omim FROM repeat_expansion WHERE id=" + id).toString().split(",");
-		for(int i=0; i<disease_ids_omim.count(); ++i)
-		{
-			QString id = disease_ids_omim[i].trimmed();
-			if (id.isEmpty()) continue;
-			QStringList names;
-			SqlQuery query = db.getQuery();
-			query.exec("SELECT phenotype FROM omim_phenotype WHERE phenotype LIKE '%" + id + "%'");
-			while (query.next())
-			{
-				QString name = query.value(0).toString();
-				name.replace(id, "");
-				name.replace("(3)", "");
-				name = name.trimmed();
-				while(name.endsWith(',') || name.endsWith(' ')) name.chop(1);
-				if (!name.isEmpty()) names << name;
-			}
-			if (!names.isEmpty())
-			{
-				disease_ids_omim[i] =  id + " - " + names.join("/");
-			}
-		}
-		setCell(row, "OMIM disease IDs", disease_ids_omim.join("; "));
-
-		//comments
-		QString comments = db.repeatExpansionComments(id.toInt());
-		QTableWidgetItem* item = setCell(row, "comments", "");
-		if (!comments.isEmpty())
-		{
-			setCellDecoration(row, "comments", comments);
-			item->setIcon(QIcon(":/Icons/Info.png"));
-		}
-
-		//HPO terms
-		QStringList hpo_terms = db.getValue("SELECT hpo_terms FROM repeat_expansion WHERE id=" + id).toString().split(",");
-		for(int i=0; i<hpo_terms.count(); ++i)
-		{
-			QString id = hpo_terms[i].trimmed();
-			QString name = db.getValue("SELECT name FROM hpo_term WHERE hpo_id LIKE '" + id + "'", true).toString();
-			if (!name.isEmpty()) id += " - " + name;
-			hpo_terms[i] = id;
-		}
-		setCell(row, "HPO terms", hpo_terms.join("; "));
-
-		//type
-		QString type = db.getValue("SELECT type FROM repeat_expansion WHERE id=" + id).toString().trimmed();
-		setCell(row, "type", type);
-
-		//type
-		bool inhouse_testing = db.getValue("SELECT inhouse_testing FROM repeat_expansion WHERE id=" + id).toBool();
-		setCell(row, "in-house testing", inhouse_testing ? "yes" : "no");
-
-		//statistical cutoff
-		if (!sys_type_cutoff_col_.isEmpty())
-		{
-			QString cutoff = db.getValue("SELECT "+sys_type_cutoff_col_+" FROM repeat_expansion WHERE id=" + id).toString().trimmed();
-			setCell(row, "statistical cutoff", cutoff);
-		}
-	}
+    NGSD db;
+    SqlQuery re_query = db.getQuery();
+    re_query.exec("SELECT * FROM repeat_expansion");
+    QList<QMap<QString, QString>> re_table_data;
```

Comment:
why no DBTable?

Reply:
It was easier to use, I guess. Can I use the filtering from DBTable to find id by region and repeat_unit?

Reply:
You can use the same loop as you do now.
You just have to determine the column indices before the loop.